### PR TITLE
make compatible iOS 15 #31

### PR DESCRIPTION
**Pull request by vijay-kachhadiya**

### CODE REVIEW

File: ZusExample/Utils/Extension/PHAsset.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          }
+          
+          // According to the docs, the image is guaranteed at this point.
+          continuation.resume(returning: image)
+        }
+      }
+    }
+  }
+  
+  func requestVideo(
```
Comment (aaryankotharii) [inline ZusExample/Utils/Extension/PHAsset.swift:70]: remove video

File: ZusExample/Utils/Extension/PHAsset.swift [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    
+    options.predicate = NSPredicate(
+      format: "mediaType = %d OR mediaType = %d",
+      PHAssetMediaType.video.rawValue,
+      PHAssetMediaType.image.rawValue
+    )
+    return options
+  }
+  
+  class var recentOptions: PHFetchOptions {
```
Comment (aaryankotharii) [inline ZusExample/Utils/Extension/PHAsset.swift:132]: remove recentOptions variable

File: ZusExample/Vult/View/AllocationActionStack.swift [hunk 3]
```diff
@@ -11,16 +11,24 @@ import PhotosUI
 struct AllocationActionStack: View {
     @EnvironmentObject var vultVM: VultViewModel
     @Environment(\.colorScheme) var colorScheme
+    @State private var isShowingPicker = false
```
Comment (aaryankotharii) [inline ZusExample/Vult/View/AllocationActionStack.swift:14]: add var isShowingPicker to VultViewModel

**Review by aaryankotharii — CHANGES_REQUESTED** [review]

File: ZusExample/Utils/Extension/PHAsset.swift (hunk 1 above)
Comment (vijay-kachhadiya) [reply ZusExample/Utils/Extension/PHAsset.swift:70]: Done

File: ZusExample/Utils/Extension/PHAsset.swift [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  class var options: PHFetchOptions {
+    let options = PHFetchOptions()
+    options.sortDescriptors = [NSSortDescriptor(
+      key: "creationDate",
+      ascending: false
+    )]
+    
+    options.predicate = NSPredicate(
+      format: "mediaType = %d OR mediaType = %d",
+      PHAssetMediaType.video.rawValue,
```
[the remark this answers is not in the document]
Comment (vijay-kachhadiya) [reply ZusExample/Utils/Extension/PHAsset.swift:126]: Done

File: ZusExample/Utils/Extension/PHAsset.swift (hunk 2 above)
Comment (vijay-kachhadiya) [reply ZusExample/Utils/Extension/PHAsset.swift:132]: Done

File: ZusExample/Vult/View/AllocationActionStack.swift [hunk 5]
```diff
@@ -11,16 +11,24 @@ import PhotosUI
...
     var body: some View {
         HStack(spacing:10) {
             
-            PhotosPicker(
-                selection: $vultVM.selectedPhotos,
-                matching: .images,
-                photoLibrary: .shared()) {
-                    WalletActionBlock(icon: "photo",title: "Upload Image")
-                }
+               /* PhotosPicker(
```
[the remark this answers is not in the document]
Comment (vijay-kachhadiya) [reply ZusExample/Vult/View/AllocationActionStack.swift:18]: Done

File: ZusExample/Vult/View/AllocationActionStack.swift (hunk 3 above)
Comment (vijay-kachhadiya) [reply ZusExample/Vult/View/AllocationActionStack.swift:14]: Done

File: ZusExample/Vult/View/AllocationActionStack.swift [hunk 6]
```diff
@@ -11,16 +11,24 @@ import PhotosUI
...
+                    matching: .images,
+                    photoLibrary: .shared()) {
+                        WalletActionBlock(icon: "photo",title: "Upload Image")
+                    }*/
             
+            WalletActionBlock(icon: "photo",title: "Upload Image")
+                .onTapGesture {
+                    isShowingPicker = true
+                }
+                .sheet(isPresented: $isShowingPicker) {
```
[the remark this answers is not in the document]
Comment (vijay-kachhadiya) [reply ZusExample/Vult/View/AllocationActionStack.swift:29]: Done

File: ZusExample/Vult/ViewModel/VultViewModel.swift [hunk 7]
```diff
@@ -51,14 +53,36 @@ class VultViewModel: NSObject, ObservableObject {
...
+                let fetchResult = PHAsset.fetchAssets(withLocalIdentifiers: identifiers, options: nil)
+                for index in 0..<fetchResult.count {
+                  let asset = fetchResult.object(at: index)
+                    do {
+                    let image = try await asset.requestImage()
+                      if let data = image.pngData() {
+                          //try uploadFile(data: data, name: name)
+                          var file = File()
+                          file.name = asset.value(forKey: "filename") as? String ?? Date().timeIntervalSince1970.description
+                          file.path = "/\(asset.value(forKey: "filename") as? String ?? Date().timeIntervalSince1970.description)"
```
[the remark this answers is not in the document]
Comment (vijay-kachhadiya) [reply ZusExample/Vult/ViewModel/VultViewModel.swift]: Done

**Pull request merged**